xds/src/main/java/io/grpc/xds/internal/sds/ClientSslContextProviderFactory.java

```diff
@@ -74,8 +74,8 @@ public SslContextProvider create(UpstreamTlsContext upstreamTlsContext) {
             upstreamTlsContext,
             bootstrapper.readBootstrap().getNode().toEnvoyProtoNode(),
             bootstrapper.readBootstrap().getCertProviders());
```

Comment:
Btw, this API usage looks bad...  @sanjaypujare

Reply:
I will fix that to remove redundant calls to `readBootstrap()`. What do you think of the idea to cache `BootstrapInfo` in the `Bootstrapper` instance and not re-read if the cached value is present?

Reply:
No, that's problematic. Remember we initially implemented this with global caching of `BootstrapInfo`, but then realized that each channel should read the bootstrap file once as each channel creates its own XdsClient. This is also how other languages implement it. In the future (e.g., sharing XdsClient), we don't know yet.

Reply:
There is a single bootstrap file for the whole application and all instances of the `XdsClient` (whether used on the channel side or the server side) should see the same content AFAIK. Why do you want each channel to re-read the bootstrap file unless you want to support the use-case of the bootstrap file getting updated during the lifetime of the application - which I think we don't want to support.

Reply:
> unless you want to support the use-case of the bootstrap file getting updated during the lifetime of the application

Yes, I remember this was the reason that we changed the initial version of implementation (with caching) to current one. But any way, the it should not affect how callers use this API as this is an implementation choice. That's why I am saying repeatedly calling `readBoostrap` is a bad practice instead of it's wrong.

Reply:
Yes, sounds good.